include/boost/safe_float/policy/cast_from.hpp

```diff
@@ -0,0 +1,69 @@
+#ifndef BOOST_SAFE_FLOAT_CAST_FROM_HPP
+#define BOOST_SAFE_FLOAT_CAST_FROM_HPP
+
+#include <type_traits>
+
+
+namespace boost {
+namespace safe_float {
+namespace policy {
+
+template<typename T>
+struct allow_cast_from {
+    
+    template<typename SF>
+    struct policy
+    {
+        template<typename U>
+        static constexpr bool can_cast_from = std::is_same_v<T, U>;
+
+        template<typename U>
+        static constexpr bool can_explicitly_cast_from = false;
```

Comment:
We want to allow implicit and not explicit? In general things that are implicitly castable are also explicitly, but not the other way around. How you enforce that only implicit is allowed?

Reply:
This is relevant when looking at the constructors in safe_float code. can_cast_from enable/disable the implicit constructor while can_explicitly_cast_from enable/disable the explicit constructor.
If only can_cast_from is true, only the implicit constructor will be enabled and can be used like usually, implicitly or explicitly.
But if both can_cast_from and can_explicitly_cast_from are true, both constructor will be enabled and it will be a problem when performing an explicit construction as the compiler will have two possible constructors to use and won't know which one to choose resulting in a compile error.
So what this boolean means really is "explicit only"
It mean that both boolean should not true at the same time, perhaps i can change the condition to enable the explicit constructor in safe_float to remove this limitation (enabling it when can_explicitly_cast_from is true and can_cast_from is false)